src/t8_forest/t8_forest_iterate.cxx

```diff
@@ -420,15 +428,134 @@ t8_forest_iterate_replace (t8_forest_t forest_new,
     T8_ASSERT (eclass == t8_forest_get_tree_class (forest_old, itree));
     ts = t8_forest_get_eclass_scheme (forest_new, eclass);
     T8_ASSERT (ts == t8_forest_get_eclass_scheme (forest_new, eclass));
+
+    /* Refine flag of -3 is not defined. Its set here for testing reasons.
+     * later in */
+    refine = -3;
+
+    T8_ASSERT(elems_per_tree_old > 0 && elems_per_tree_new > 0);
+    ts->t8_element_new(1, &elem_parent);
+  
     for (ielem_new = 0, ielem_old = 0; ielem_new < elems_per_tree_new
-         || ielem_old < elems_per_tree_old;) {
+        || ielem_old < elems_per_tree_old; ) {
+    
+      T8_ASSERT(ielem_new <= elems_per_tree_new);
+      T8_ASSERT(ielem_old <= elems_per_tree_old);
+
       /* Iterate over the elements */
       /* Get pointers to the elements */
       elem_new = t8_forest_get_element_in_tree (forest_new, itree, ielem_new);
       elem_old = t8_forest_get_element_in_tree (forest_old, itree, ielem_old);
+      
       /* Get the levels of these elements */
       level_new = ts->t8_element_level (elem_new);
       level_old = ts->t8_element_level (elem_old);
+
+      if (level_old < level_new) {
+        /* elem_old may was refined */
+        ts->t8_element_parent(elem_new, elem_parent);
+        if (!ts->t8_element_compare (elem_old, elem_parent)){
+          /* elem_old was refined */
+          T8_ASSERT (level_new == level_old + 1);
+          refine = 1;
+          family_size = ts->t8_element_num_children (elem_old);
+#if T8_DEBUG
+          /* Check if family of new refined elements is complete */
+          T8_ASSERT (ielem_new + family_size <= elems_per_tree_new);
+          for (ielem = 1; ielem < family_size; ielem++) {
+            elem_new = t8_forest_get_element_in_tree (forest_new, itree, 
+                        ielem_new + ielem);
+            ts->t8_element_parent(elem_new, elem_parent);
+            SC_CHECK_ABORT (ts->t8_element_compare (elem_old, elem_parent),
+                  "Family is not complete");
+          }
+#endif
+          t8_debugf("[IL] itree %i, elem_old %i, elem_new %i, refine %i, family_size 1\n", 
```

Comment:
Please move it into the debug-makros, or make it proper-debugging output (Remove the [IL] at the beginning)